src/sqlfluff/utils/reflow/elements.py

```diff
@@ -497,13 +523,128 @@ def _handle_respace__inline_without_space(
 
         return segment_buffer, existing_fixes + new_fixes, edited
 
+    def _get_indent_segment(self) -> Optional[RawSegment]:
+        """Get the current indent segment (if there)."""
+        for seg in reversed(self.segments):
+            if seg.is_type("newline", "end_of_file"):
+                return None
+            elif seg.is_type("whitespace"):
+                return seg
+        return None
+
+    def get_indent(self) -> Optional[str]:
+        """Get the current indent (if there)."""
+        seg = self._get_indent_segment()
+        return seg.raw if seg else None
+
+    def indent_to(
+        self,
+        desired_indent: str,
+        after: Optional[BaseSegment] = None,
+        before: Optional[BaseSegment] = None,
+    ) -> Tuple[List[LintFix], "ReflowPoint"]:
+        """Coerce a point to have a particular indent.
+
+        If the point currently contains no newlines, one will
+        be introduced and any trailing whitespace will be effectively
+        removed.
+
+        More specifically, the newline is _inserted_ before the existing
+        whitespace, with the new indent being a replacement for that
+        same whitespace.
+        """
+        # Get the indent (or in the case of no newline, the last whitespace)
+        indent_seg = self._get_indent_segment()
+        reflow_logger.debug(
+            "Coercing indent %s to %r. (newlines: %s)",
+            indent_seg,
+            desired_indent,
+            self.num_newlines(),
+        )
+        if self.num_newlines():
+            # There is already a newline.
+            if indent_seg:
+                # Coerce existing indent to desired.
+                if indent_seg.raw == desired_indent:
+                    # Trivial case. Indent already correct
+                    return [], self
+                elif desired_indent == "":
+                    # Coerce to no indent. We don't want the indent. Delete it.
+                    new_indent = indent_seg.edit(desired_indent)
+                    idx = self.segments.index(indent_seg)
+                    return [LintFix.delete(indent_seg)], ReflowPoint(
+                        list(self.segments[:idx]) + list(self.segments[idx + 1 :])
```

Comment:
Is it necessary to put `list()` around the two slices here?

Reply:
No - this is good feedback which I should have taken from the first PR too 🤦 . I think given your other comment, and that a `ReflowElement` is a frozen `dataclass` I've instead swapped the datatype to a `tuple` and then also removed all the unnecessary type changes. There are a couple still in there now where we do the working logic in a `list` and then convert to tuple on constructing a `ReflowElement` but mostly we can work on slicing in the native type 👍